lib/rucio/core/did.py

```diff
@@ -1176,6 +1176,12 @@ def set_metadata(scope, name, key, value, type=None, did=None,
     :param recursive: Option to propagate the metadata change to content.
     :param session: The database session in use.
     """
+    try:
+        rowcount = session.query(models.DataIdentifier).filter_by(scope=scope, name=name).\
+            with_hint(models.DataIdentifier, "INDEX(DIDS DIDS_PK)", 'oracle').one()
+    except NoResultFound:
+        raise exception.DataIdentifierNotFound("Data identifier '%s:%s' not found" % (scope, name))
+
```

Comment:
I did it that way so that it always returns a `DataIdentifierNotFound` if the DID doesn't exist. Otherwise, it can raise an other exception if the DID doesn't exist AND a wrong metadata is defined (e.g. TypeError line 1185)